pkg/config/schema.go

```diff
@@ -126,38 +148,61 @@ func validateRange(path string, limit interface{}, data interface{}, isMinimum b
 // Initialize sets up the schema
 func (check *SchemaCheck) Initialize(id string) error {
 	check.ID = id
-	if check.JSONSchema != "" {
-		if err := json.Unmarshal([]byte(check.JSONSchema), &check.Schema); err != nil {
+	if check.SchemaString == "" {
+		jsonBytes, err := json.Marshal(check.Schema)
+		if err != nil {
 			return err
 		}
+		check.SchemaString = string(jsonBytes)
 	}
-	return nil
+	err := json.Unmarshal([]byte(check.SchemaString), &check.Validator)
+	return err
+}
+
+// TemplateForResource fills out a check's templated fields given a particular resource
+func (check SchemaCheck) TemplateForResource(res interface{}) (*SchemaCheck, error) {
+	newCheck := check
```

Comment:
Why create another variable for the check?

Reply:
Good question! It's actually making a copy so we can safely modify the schema. Added a comment since that's non-obvious.